src/arm/windows/init.c

```diff
@@ -0,0 +1,257 @@
+		NULL, /* Request buffer size */
+		&dataSize);
+	if (result != 0 || dataSize == 0) {
+		cpuinfo_log_error("Registry entry size read error");
+		return false;
+	}
+
+	if (*textBuffer) {
+		HeapFree(heap, 0, *textBuffer);
+	}
+	*textBuffer = HeapAlloc(heap, HEAP_ZERO_MEMORY, dataSize);
+	if (*textBuffer == NULL) {
+		cpuinfo_log_error("Registry textbuffer allocation error");
+		return false;
+	}
+
+	result = RegGetValue(
+		HKEY_LOCAL_MACHINE,
+		subkey,
+		value,
+		flags,
+		NULL,
+		*textBuffer, /* Write string in this destination buffer */
+		&dataSize);
+	if (result != 0) {
+		cpuinfo_log_error("Registry read error");
+		return false;
+	}
+	return true;
+}
+
+static bool get_system_info_from_registry(
+	struct woa_chip_info** chip_info,
+	enum cpu_info_vendor* vendor)
+{
+	DWORD keyType = 0;
+	DWORD dataSize = 0;
+	const DWORD flags = RRF_RT_REG_SZ; /* Only read strings (REG_SZ) */
+	bool result = false;
+	char* textBuffer = NULL;
+	LPCTSTR cpu0_subkey =
+		"HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0";
+	LPCTSTR chip_name_value = "ProcessorNameString";
+	LPCTSTR vendor_name_value = "VendorIdentifier";
+
+	*chip_info = NULL;
+	*vendor = cpuinfo_vendor_unknown;
+	HANDLE heap = GetProcessHeap();
+
+	/* 1. Read processor model name from registry and find in the hard-coded list. */
+	if (!read_registry(cpu0_subkey, chip_name_value, &textBuffer)) {
+		cpuinfo_log_error("Registry read error");
+		goto cleanup;
+	}
+	for (uint32_t i = 0; i < (uint32_t) woa_chip_name_last; i++) {
+		size_t compare_length = strnlen(woa_chips[i].chip_name_string, CPUINFO_PACKAGE_NAME_MAX);
+		int compare_result = strncmp(textBuffer, woa_chips[i].chip_name_string, compare_length);
+		if (compare_result == 0) {
+			*chip_info = woa_chips+i;
+		}
+	}
+	if (*chip_info == NULL) {
+		cpuinfo_log_error("Unknown chip model name.\n Please add new Windows on Arm SoC/chip support!");
+		goto cleanup;
+	}
+	cpuinfo_log_debug("detected chip model name: %s", (**chip_info).chip_name_string);
+
+	/* 2. Read vendor/manufacturer name from registry. */
+	if (!read_registry(cpu0_subkey, vendor_name_value, &textBuffer)) {
+		cpuinfo_log_error("Registry read error");
+		goto cleanup;
+	}
+
+	for (uint32_t i = 0; i < (sizeof(vendors) / sizeof(struct vendor_info)); i++) {
+		if (strncmp(textBuffer, vendors[i].vendor_name,
+				strlen(vendors[i].vendor_name)) == 0) {
+			*vendor = vendors[i].cpu_info_vendor;
+		}
+		result = true;
+	}
+	if (*vendor == cpuinfo_vendor_unknown) {
+		cpuinfo_log_error("Unexpected vendor: %s", textBuffer);
+	}
+
+cleanup:
+	if (textBuffer) {
+		HeapFree(heap, 0, textBuffer);
+	}
+	return result;
+}
+
+static void get_isa()
+{
+	bool armv8 = IsProcessorFeaturePresent(PF_ARM_V8_INSTRUCTIONS_AVAILABLE);
+	bool crypto = IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+	bool load_store_atomic = IsProcessorFeaturePresent(PF_ARM_64BIT_LOADSTORE_ATOMIC);
+	bool divide = IsProcessorFeaturePresent(PF_ARM_DIVIDE_INSTRUCTION_AVAILABLE);

```

Comment:
there are a bunch of processor feature flags that you retrieve and are unused besides the debug log. 
Can you remove everything that is not used or move it into a #ifdef debug block with the log line.